var/spack/repos/builtin/packages/intel-oneapi-compilers-classic/package.py

```diff
@@ -21,15 +20,20 @@ class IntelOneapiCompilersClassic(Package):
 
     phases = []
 
-    for ver in ['2022.1.0',
-                '2022.0.2',
-                '2022.0.1',
-                '2021.4.0',
-                '2021.3.0',
-                '2021.2.0',
-                '2021.1.2']:
+    # Versions before 2021 are in the `intel` package
+    # intel-oneapi versions before 2022 use intel@19.0.4
+    for ver, oneapi_ver in {
+        '2021.6.0': '2022.2.0',
+        '2021.5.0': '2022.1.0',
+    }.items():
         version(ver)
-        depends_on('intel-oneapi-compilers@' + ver, when='@' + ver, type='run')
+        depends_on('intel-oneapi-compilers@' + oneapi_ver, when='@' + ver, type='run')
```

Comment:
This untangles the mess with having 2 compilers in the same package with different version numbers. Much better.

Reply:
Are there similar issues we need to untangle for other components, or are the other components in Spack able to be versioned properly since they don't share an installer the same way? I haven't installed things like vtune to confirm that the version it reports matches the version Spack knows about yet, thought you might know off the top of your head without needing to test it out.

Reply:
The compiler is the only one.